/*
 * Copyright(c) 2012-2018 Intel Corporation
 * SPDX-License-Identifier: BSD-3-Clause-Clear
 */

syntax = "proto3";
option cc_generic_services = true;
import "trace.proto";
package octf.proto;

message Distribution {
    string unit = 1;

    uint64 avarage = 2;

    uint64 min = 3;

    uint64 max = 4;

    uint64 count = 5;

    uint64 total = 6;

    map<string, uint64> histogram = 100;

    map<string, uint64> percentile = 101;
}

message IoStatisticsEntry {
    Distribution size = 10;

    Distribution latency = 11;
}

message IoStatisticsDescription {
    trace.EventDeviceDescription devcie = 1;
}

message IoStatistics {
    IoStatisticsDescription desc = 1;

    IoStatisticsEntry read = 101;

    IoStatisticsEntry write = 102;

    IoStatisticsEntry discard = 103;

    IoStatisticsEntry total = 104;
}

message IoStatisticsSet {
    repeated IoStatistics statistics = 1;
}
